crates/generator/src/generator.rs

```diff
@@ -123,22 +190,28 @@ impl Generator {
         &self.account_lock_manage
     }
 
-    #[instrument(skip_all, fields(backend = ?backend.backend_type))]
-    fn machine_run<'a, S: State + CodeStore, C: ChainView>(
-        &'a self,
-        chain: &'a C,
-        state: &'a S,
-        block_info: &'a BlockInfo,
-        raw_tx: &'a RawL2Transaction,
-        max_cycles: u64,
-        backend: Backend,
+    #[instrument(skip_all, fields(backend = ?args.backend.backend_type))]
+    fn machine_run<S: State + CodeStore, C: ChainView>(
+        &self,
+        args: MachineRunArgs<'_, C, S>,
     ) -> Result<RunResult, TransactionError> {
         const INVALID_CYCLES_EXIT_CODE: i8 = -1;
 
+        let MachineRunArgs {
+            chain,
+            state,
+            block_info,
+            raw_tx,
+            max_cycles,
+            backend,
+            cycles_pool,
+        } = args;
+
         self.redir_log_handler.start(raw_tx);
         let mut run_result = RunResult::default();
-        let used_cycles;
+        let mut used_cycles;
         let exit_code;
+        let cycles_pool_bak = cycles_pool.clone();
```

Comment:
Use the `origin_cycle_meter` or `cycle_meter_before_execute` to make the code easy to read.